### DISCUSSION TITLE
Balkonsolar 3x MPPT / Pylontech / Hoymiles HMS 1600

### DISCUSSION THREAD

**Discussion by Mercury0802** [hoylabs/OpenDTU-OnBattery#1276] [Q&A]
Hallo zusammen,

erstmal vielen Dank für das mega Projekt und die aktive Community! 
Habe bei mir schon seit längerem eine Balkonsolaranlage mit openDTU am laufen, die jetzt etwas geupgraded wurde. 

Kurz zu den Eckdaten:
2x Victron MPPT 150/35
1x Victron MPPT 250/60 (leider nötig wegen 3er Strings mit anderen Modulen, die im Worst Case über 150V liegen)
Umstellung von ESP32 auf ESP32S3 (vor allem wegen des zusätzlichen UARTS)
US5000 4,8kwh
Nulleinspeisung, keine Full Solar Passthrough (nur Passthrough oberhalb der VBatt-Stop Schwelle - find ich richtig gut)
WR: Hoymiles 1600 4T
Leistungsmessung über Volkszähler (Wert wird bei jeder Veränderung bereitsgestellt. - Über HA dann mit mqtt gepublisht). Verbrauch wird richtig gut ausgeregelt, bis auf die bekannten Szenarien in der Lastdynamik, worüber man hier im Forum sonst auch so stolpert.

Ein Thema beschäftigt mich gerade noch, worüber ich hier im Forum auch schon gelesen habe.
[Kommunikation Victrons](https://github.com/helgeerbe/OpenDTU-OnBattery/pull/739)

Seit der Umstellung auf den ESP32S3 sehe ich alle 3 MPPTs + Pylontech CAN.

Hintergrund meiner nachfolgenden Frage ist, dass bei entsprechendem SOC (i.d.R. > 90%) der maximale Ladestrom auf 20A begrenzt wird. Der vom BMS zurückgemeldete Ladestrom liegt in den bisherigen Messungen drüber (ca. 8-10A). Bisher aber keine Warnings oder Alarme.
Im Grunde macht mir das auch keine zu großen Sorgen, weil ja durch die [Victron & Pylontech Reglereinstellungen](https://www.victronenergy.com/live/battery_compatibility:pylontech_phantom) der Laderegler irgendwann in der Absorption landet. Wenn das BMS allerdings eine Grenze festlegt, sollte man diese auch einhalten.

Meine Fragen sind:
Ist es Stand heute implementiert, den maximalen Ladestrom der Victrons über die RX/TX Schnittstelle zu begrenzen? 
Falls ja:
- Kann es sein, dass der Laderegler, der mir keine Temp und VE. Gesamtleistung anzeigt, falsch angepinnt ist und deshalb die Begrenzung nicht mitbekommt?
- Wie kann ich prüfen, ob ein Laderegler dieses Limit mitbekommt und wie werden bei 3 ggf. unterschiedlichen MPPTs die einzelnen Grenzen festgelegt?

Würde es Sinn machen, die Absorptionsspannung ggf. weiter zu reduzieren, um ggf. hohe Ladeströme zu vermeiden?

Mit meiner Anlage kommen max. 50A@48V Peak an einem sonnigen Tag.

Sorry, wahrscheinlich gibt's für alle Fragen schon irgendwo Antworten hier, aber außer 20 neuer Lesezeichen ist mein Informationsgewinn nicht so gut gelaufen.

Danke!

**Comment by schlimmchen** [comment]
> Ist es Stand heute implementiert, den maximalen Ladestrom der Victrons über die RX/TX Schnittstelle zu begrenzen?

Nein.

**Comment by Manos1966** [comment]
Ich will davon ausgehen, dass deine drei Victrons MPPT 150/35 & MPPT 250/60 Smartsolar sind und miteinander kommunizieren, richtig?
Falls nicht, unbedingt  [Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Victron-MPPT-solar-charger#using-more-than-one-victron-mppt-with-opendtu-onbattery)

50A Ladestrom ist laut Pylontech US5000 unproblematisch, oder steht anderes in deiner Bedienungsanleitung?
![Capture](https://github.com/user-attachments/assets/94309a8e-077b-4819-96ec-5331fb72de16)

**Comment by Mercury0802** [comment]
Ja die 3 MPPTs sind im VE.Networking und arbeiten kollaborativ. Das passt soweit auch. Die Laderegler ändern ihren Status zeitgleich von z.B. Bulk -> Absorption.
50A Ladestrom sind auch in weiten Betriebsbereichen in Ordnung, aber bei hohen SOCs wird die Ladeleistung reduziert (LiFePo4 und NMC verhalten sich da ähnlich). Ein PKW der mit 200kW laden kann, macht das auch nicht bis der Akku voll ist.
Bin zwar kein Batterieexperte aber bei Pylontech hast du einen passiven Balancer, der üblicherweise eher am Ende des Ladevorgangs die Zellspannungen versucht auszugleichen. Das benötigt Zeit und wird bei vergleichsweise niedrigen Ladeleistungen gemacht.
Hier ein Screenshot der die BMS Signale zeigt: (SOC 98%)

![BMS_Limit_current_high_SOC](https://github.com/user-attachments/assets/84314397-8f80-4e54-9a8f-2a6df0219ad7)

**Reply by Manos1966** [reply]
> 50A Ladestrom sind auch in weiten Betriebsbereichen in Ordnung, aber bei hohen SOCs wird die Ladeleistung reduziert 

Du beschreibst die Hauptaufgabe des Victron MPPT 
Da hat die OpenDTU-OnBattery wenig damit zu tun AUSSER der Option "Full Solar Passthrough" um den Batterie Stand auf einen beliebigen Wert (< 100%) zu halten (SUPER FUNKTION! 👍 )

Deine Victrons sollten diese Aufgabe selbststaendig meistern. Die 50A sind nur 50% Prozent der US5000 Faehigkeit, also hier sollten es kein Problem sein.
Und ich gehe davon aus, du hast auch das hier gemacht (aus dem Victron Pylontech Handbuch):

> The Pylontech battery has 15 cells in series, so 53.2V equates to 3.55V per cell. This is very highly charged and makes the system prone to go overvoltage.
It should also be noted that a LiFePO4 cell stores very little additional energy above 3.45V.
**For this reason we opted to override the BMS and cap the voltage at 52.4V.** This sacrifices almost none of the capacity and greatly improves the stability of the system.

PS. Du koenntest sogar die Ladespannung unter 52,4V stellen, Dann wird zwar deine Batterie etwas langsamer geladen, dafuer aber schonender und nicht bis zu 100% jedes Mal. Anbei indikativ die Spannung und entsprechender Ladestand (SOC)%
![00 Spannungen der Pylontech US-Serie](https://github.com/user-attachments/assets/b7629d08-a3a4-4849-ba29-b051d5d222eb)

**Reply by Mercury0802** [reply]
Kann in meinen Augen nicht die Funktion des MPPT sein, da eine Spannungsregelung (und damit Ladestromlimitierung) erst ab "Absorption" stattfindet. In dieser Konstellation wie z.B.
[ExampleA](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Victron-150-35-MPPT,-100-20-MPPT,-9x405Wp,-Pylontech-4,8kWh-48V,-HM%E2%80%90600)
[Example B](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/360)
könnte ich mir ebenfalls genau so einen Fall vorstellen. Dort sind auch Ströme > 20A möglich.

Ich kann mir gut vorstellen, dass man dort auch in die Situation kommen kann, dass der Ladestrom größer ist als der maximal zulässige Ladestrom.

Laut Victron ist für den Betrieb mit der Pylontech ein Cerbo GX o.ä. für die Reaktion auf Stromgrenzen nötig:

![Victron_1 2](https://github.com/user-attachments/assets/b80f005a-7c8c-4a6c-8e73-414dbd300e6c)

Die Ladeschlussspannung = Absorptionsspannung steht bei mir ohnehin auf dem offiziell von Victron empfohlenen Wert (52,0V)
[siehe 5.2](https://www.victronenergy.com/live/battery_compatibility:pylontech_phantom)

Die Ladeschlussspannung und der zulässige Strom sind zwei völlig unterschiedliche Dinge. Wenn ein BMS eine oder mehrere Zellen mit hoher Spannung erkennt, soll diese Zelle nicht durch eine hohe Ladeleistung weiter geladen werden. Dann wird das BMS aktiv und gibt einen maximalen Ladestrom vor.
Es gibt auch in meinen Augen keinen nachvollziehbaren Grund, diese Ladestrombegrenzung nicht ernst zu nehmen.

Klar könnte man jetzt auch sagen, was soll passieren, das BMS kann bei Overvoltage, Overcurrent usw. abschalten. Ob das im Sinne der Langlebigkeit und dauerhaft ohne Zellschädigung gut geht?

Full Solar Passthrough habe ich mir auch schon überlegt. Wäre sicherlich eine Option um die Ladespannung herunterzusetzen, aber nicht jeder möchte ins Netz einspeisen. Außerdem kommt man dann eben nicht mehr in Absorption und Float, um den Ladezyklus sauber abzuschließen.

**Reply by Manos1966** [reply]
>  Wenn ein BMS eine oder mehrere Zellen mit hoher Spannung erkennt, soll diese Zelle nicht durch eine hohe Ladeleistung weiter geladen werden. Dann wird das BMS aktiv und gibt einen maximalen Ladestrom vor.
Es gibt auch in meinen Augen keinen nachvollziehbaren Grund, diese Ladestrombegrenzung nicht ernst zu nehmen.

Das ist der Status meiner Pylontechs (es sind zwei)
160A Charge Rate / 2 = 80A Charge Rate pro Batterie.
![Capture](https://github.com/user-attachments/assets/968dfa9e-ea67-4753-8958-9310a385c0e3)

IN WIRKLICHKEIT die Pylontechs sind 0,5C Batterien. Fuer die US5000 wuerde es bedeuten 100Ah x 0,5 = 50A continous charging.
Dein System gibt bestenfalls 50A, damit bist genau am 0,5C Limit und du kannst bist zum Ende mit 0,5C die Batterie laden. Den Rest am Ende, regelt der Victron MPPT automatisch.

Wenn (Falls) deine Batterie Zellen Probleme haben, das ist eine andere Baustelle... Eigentlich hast du deswegen Pylontech gekauft (um diese Probleme nicht zu haben) aber wenn du dir solche Sorgen machts, dann brauchst du Zugriff auf die RS485 Schnittstelle und eine Zellen-Ueberwachung.
Ich habe eine solche Loesung Parallel-laufen zu meiner OpenDTU-OnBattery, aber es ist nur "Just for fun".  Siehe [RS485](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/622#discussioncomment-8726163)
In Wirklichkeit, der Gesundheitsstand % deiner Batterie wird dir eine Indikation geben wenn Probleme auftauchen.

**Comment by hhoebel** [comment]
Ich habe ein vergleichbares Problem - 3 Victron-MPPT mit maximal 135A Ladestrom und eine Batterie, die zwar bis zu 210A kann, aber gerade bei niedrigen SOC gerne auch mal eine Ladestromlimit von nur 60A per CAN empfiehlt (Delongtop-Akkus mit Pace-BMS, das das Pylontech-Protokoll beherrscht). Da ich diese Vorgabe auch gerne einhalten würde, habe ich auf Basis des Release vom 23-01-25 die entsprechende Funktionalität mal lokal eingepflegt und getestet - siehe auch die dazu passende Diskussion hier: https://github.com/hoylabs/OpenDTU-OnBattery/issues/470#issuecomment-2633212493
